pom.xml

```diff
@@ -36,68 +36,179 @@
     </dependency>
 
     <dependency>
       <groupId>org.apache.hive</groupId>
       <artifactId>hive-service</artifactId>
       <version>${hive.version}</version>
+      <exclusions>
+      	<exclusion>
+      	  <groupId>log4j</groupId>
+      	  <artifactId>log4j</artifactId>
+      	</exclusion>
+      	<exclusion>
+      	  <groupId>org.slf4j</groupId>
+      	  <artifactId>slf4j-log4j12</artifactId>
+      	</exclusion>
+      	<exclusion>
+          <groupId>org.pentaho</groupId>
+          <artifactId>pentaho-aggdesigner-algorithm</artifactId>
+        </exclusion>
+      </exclusions>
     </dependency>
 
     <dependency>
       <artifactId>tez-dag</artifactId>
       <groupId>org.apache.tez</groupId>
       <version>${tez.version}</version>
+      <exclusions>
+      	<exclusion>
+      	  <groupId>log4j</groupId>
+      	  <artifactId>log4j</artifactId>
+      	</exclusion>
+      	<exclusion>
+      	  <groupId>org.slf4j</groupId>
+      	  <artifactId>slf4j-log4j12</artifactId>
+      	</exclusion>
+      </exclusions>
     </dependency>
 
     <dependency>
       <artifactId>tez-common</artifactId>
       <groupId>org.apache.tez</groupId>
       <version>${tez.version}</version>
+      <exclusions>
+      	<exclusion>
+      	  <groupId>log4j</groupId>
+      	  <artifactId>log4j</artifactId>
+      	</exclusion>
+      	<exclusion>
+      	  <groupId>org.slf4j</groupId>
+      	  <artifactId>slf4j-log4j12</artifactId>
+      	</exclusion>
+      </exclusions>
     </dependency>
 
     <dependency>
       <artifactId>tez-mapreduce</artifactId>
       <groupId>org.apache.tez</groupId>
       <version>${tez.version}</version>
+      <exclusions>
+      	<exclusion>
+      	  <groupId>org.slf4j</groupId>
+      	  <artifactId>slf4j-log4j12</artifactId>
+      	</exclusion>
+      </exclusions>
     </dependency>
 
     <dependency>
       <groupId>org.apache.hive</groupId>
       <artifactId>hive-contrib</artifactId>
       <version>${hive.version}</version>
       <scope>test</scope>
+      <exclusions>
+        <exclusion>
+          <groupId>org.pentaho</groupId>
+          <artifactId>pentaho-aggdesigner-algorithm</artifactId>
+        </exclusion>
+      </exclusions>
     </dependency>
 
     <dependency>
       <groupId>org.reflections</groupId>
       <artifactId>reflections</artifactId>
       <version>0.9.8</version>
     </dependency>
+    
+    <dependency>
+      <groupId>org.apache.logging.log4j</groupId>
+      <artifactId>log4j-slf4j-impl</artifactId>
+      <version>${log4j2.version}</version>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.logging.log4j</groupId>
+      <artifactId>log4j-1.2-api</artifactId>
+      <version>${log4j2.version}</version>
+    </dependency>
+    <dependency>
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-api</artifactId>
+      <version>1.7.21</version>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.logging.log4j</groupId>
```

Comment:
Minor, but would be nicer if this was moved up to the other `org.apache.logging.log4j` dependencies so that they are all grouped together.